### PR TITLE
Fix installation dependencies

### LINKED ISSUE #14895: [DOC] Build from source link for ubuntu is broken
On this page: https://github.com/apache/incubator-mxnet/blob/master/docs/install/build_from_source.md

If click the ubuntu hyperlink, it is broken:
https://github.com/apache/incubator-mxnet/blob/master/docs/install/ubuntu_setup.html

### PR DESCRIPTION
**Pull request by aaronmarkham**
## Description ##
Issues related to Mac install and graphviz are addressed. 
Also updates the links in the install guides so links when browsing github will work.
Fixes #14895.

## Preview
http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-14987/3/index.html

### CODE REVIEW

**Comment by pinaraws** [comment]
@mxnet-label-bot add[CI, Doc, pr-awaiting-review]

File: docs/install/build_from_source.md [hunk 1]
```diff
@@ -58,12 +58,12 @@ MXNet's newest and most popular API is Gluon. Gluon is built into the Python bin
...
 
-* [Amazon Linux / CentOS / RHEL](centos_setup.html)
-* [macOS](osx_setup.html)
-* [Raspbian](raspian_setup.html)
-* [TX2](tx2_setup.html)
-* [Ubuntu](ubuntu_setup.html)
-* [Windows](windows_setup.html)
+* [Amazon Linux / CentOS / RHEL](centos_setup.md)
+* [macOS](osx_setup.md)
+* [Raspbian](raspian_setup.md)
```
Comment (IvyBazan) [inline docs/install/build_from_source.md]: Raspbian and TX2 links are broken

Comment (aaronmarkham) [reply docs/install/build_from_source.md]: I could fix the tx2 one, but not raspbian in a great way since that install page is pretty messed up.

**Comment by IvyBazan** [comment]
LGTM!

**Comment by wkcn** [comment]
Merged. Thank you!

**Pull request merged**